projection/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala

```diff
@@ -526,6 +526,11 @@ private[projection] class R2dbcOffsetStore(
           }
         case _ => true
       }
+    if (newInflight.size >= 10000) {
```

Comment:
added this as a precaution of any bugs resulting in inFlight not cleaned up as expected